libobs/obs-cocoa.m

```diff
@@ -889,8 +894,38 @@ bool obs_hotkeys_platform_is_pressed(obs_hotkeys_platform_t *plat,
 	return plat->is_key_down[key];
 }
 
+static void mach_set_thread_realtime()
+{
+	mach_timebase_info_data_t timebase_info;
+	mach_timebase_info(&timebase_info);
+
+	const uint64_t NANOS_PER_MSEC = 1000000ULL;
+	double clock2abs =
+		((double)timebase_info.denom / (double)timebase_info.numer) *
+		NANOS_PER_MSEC;
+
+	thread_time_constraint_policy_data_t policy;
+	policy.period = 0;
```

Comment:
I don't think 0 is a valid value for period. I'm looking at how Chromium is using this API for audio, and one code path shows:
period = 2.9 ms
computation = 75% * period
constraint = 85% * period
preemptible = NO

Maybe OBS could do similar for the graphics thread, but use the reciprocal of frame rate instead of 0 or 2.9 ms.

Reply:
Hey @jpark37 , thanks for looking into that!
These were the values as described by apple documentation but looking at the chromium source I would tend to agree.

Here is a comparison of 0 period vs 16ms period:
<img width="1163" alt="Screen Shot 2022-12-21 at 8 03 08 AM" src="https://user-images.githubusercontent.com/231018/208923751-72c5cacd-46f4-4ec8-a52d-73c136d86e9f.png">

## 0ms period:
<img width="996" alt="Screen Shot 2022-12-21 at 8 06 17 AM" src="https://user-images.githubusercontent.com/231018/208924475-e1fa299b-bf62-47b8-8622-b579b37dfd1c.png">

## 16ms period
<img width="1043" alt="Screen Shot 2022-12-21 at 8 07 19 AM" src="https://user-images.githubusercontent.com/231018/208924512-22aa5df5-7a95-4c37-9c0d-ea42c964ab4f.png">

I'll go ahead and move this to 16ms for now

Reply:
Wouldn't that lock it into supporting 60fps as a maximum though?

Reply:
It uses the obs interval now, during the draft I was hardcoding things

Reply:
I wasn't able to actually witness any demonstrable difference between different periods (16ms period with 30fps stream or 33ms period with 60fps stream).